addons/udes_delivery_control/models/picking_extra_fields.py

```diff
@@ -0,0 +1,147 @@
+from odoo import api, fields, models, _
+from ..misc import date_diff, validate_dates
+
+
+class StockPickingExtraFields(models.Model):
+    """
+        Model to hold all extra fields of stock.picking object
+    """
+
+    _name = "udes_picking_extras"
+    _description = "UDES Picking Extra Fields"
+
+    picking_id = fields.Many2one("stock.picking", required=True, index=True, copy=False)
+
+    # Unloading Fields
+    u_supplier_id = fields.Many2one(
+        "res.partner", string="Supplier", domain="[('supplier', '=', True)]"
+    )
+    u_user_id = fields.Many2one("res.users", string="User", help="Picking completed by user")
+    u_delivery_note = fields.Char("Delivery Note")
+    u_location_id = fields.Many2one("stock.location", string="Location")
+    u_vehicle_scheduled_date = fields.Datetime(
+        "Scheduled Date/Time", help="Date/Time when the vehicle is expected to arrive"
+    )
+    u_vehicle_arrival_date = fields.Datetime("Arrival Date/Time", help="Arrival time of Vehicle")
+    u_week = fields.Integer("Week", compute="_compute_u_week", store=True, default=0)
+    u_unloading_start_date = fields.Datetime("Date Start", help="Unloading start date/time")
+    u_unloading_end_date = fields.Datetime("Date End", help="Unloading completed date/time")
+    u_time_taken = fields.Float(
+        "Time Taken HH:MM",
+        compute="_compute_u_time_taken",
+        store=True,
+        readonly=True,
+        help="The amount of time taken to completeunloading in HH:MM format",
```

Comment:
Nit: Missing space between complete and unloading 👀

Reply:
done